services/ui-src/src/components/reports/DrawerReportPage.tsx

```diff
@@ -121,8 +127,48 @@ export const DrawerReportPage = ({ route, validateOnRender }: Props) => {
     onClose();
   };
 
+  const openRowDrawer = (entity: EntityShape) => {
+    setSelectedEntity(entity);
+    onOpen();
+  };
+
+  const enterButton = (entity: EntityShape, isEntityCompleted: boolean) => {
```

Comment:
The buttons aren't that different under the conditionals. Could combine them with something like this:

```
const enterButton = (entity: EntityShape, isEntityCompleted: boolean) => {
  let disabled = false;
  let style = sx.enterButton;

  if ((route.path === "/mcpar/plan-level-indicators/ilos" && !hasIlos) ||
      (route.path === "/mcpar/plan-level-indicators/prior-authorization" && isDisabled)) {
    style = sx.disabledButton;
    disabled = true;
  }

  return (
    <Button
      sx={style}
      onClick={() => openRowDrawer(entity)}
      variant="outline"
      disabled={disabled}
    >
      {isEntityCompleted ? "Edit" : "Enter"}
    </Button>
  );
};
```

Also add tests? 😀

Reply:
great call! thank you 🙏🏽 I'll add some tests now, just addressed the styling nits 😄